src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionPools.cs

```diff
@@ -39,7 +39,12 @@ public HttpConnectionPools(int maxConnectionsPerServer)
             _maxConnectionsPerServer = maxConnectionsPerServer;
             _pools = new ConcurrentDictionary<HttpConnectionKey, HttpConnectionPool>();
             // Start out with the timer not running, since we have no pools.
+
+            // Don't capture the current ExecutionContext and its AsyncLocals onto the timer causing them to live forever
+            ExecutionContext.SuppressFlow();
```

Comment:
This needs to check whether flow was already suppressed, and then only suppress and restore flow if it wasn't already.

Reply:
Fixed